### ISSUE TITLE
Nullability with tuple equality

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#37145]
**Version Used**: https://github.com/dotnet/roslyn/commit/898bed74a765fedf3071c6f25c3804995d39c7fa

**Steps to Reproduce**:

```cs
#nullable enable
using System;
public class C {
    void M(object? a, object? b) {
        _ = (a,b) != default ? a.ToString() : null; // warns
        _ = (a,b) != default ? b.ToString() : null; // warns
    }
}
```

**Expected Behavior**: no warnings

**Actual Behavior**: warns on both cases

**Comment by RikkiGibson** [comment]
It seems like this check could return `true` if one or the other or both fields are non-null, so the compiler can't assume either of the fields to be non-null. Am I missing something?

**Comment by alrz** [comment]
Yup, I tripped myself there.
